### ISSUE TITLE
Seems a command line argument bug for overwriting multiapp input parameters

### ISSUE THREAD

**Issue by YaqiWang** [idaholab/moose#5998]
which can be seen with `yak/tests/actions/neutron_diffusion_cfem:1G_multi_region_nofsrc_test` under branch `couple_reactor` of my fork.  @permcody I checked the syntax few times, it seems I am following the syntax showed in the moose tests under `tests/parser/cli_multiapp*` correctly but I did not see exodus outputs with modified names for some of multiapps.

**Comment by permcody** [comment]
Before I go digging in your fork, I assumed you checked in your CLI arguments as well?

**Comment by YaqiWang** [comment]
I did. Basically I copied the command print by test script and ran it separately not sure if it is good enough.

**Comment by YaqiWang** [comment]
@permcody and this one.

**Comment by YaqiWang** [comment]
I think we should at least fix knowing issues in the multiapp system before investigating sprint problem further.

**Comment by permcody** [comment]
I just looked for this test and I couldn't find it. I was able to check out "coupled_reactor" from 2015 and went to that folder but that test doesn't appear to exist.

**Comment by YaqiWang** [comment]
Let me take a look.

**Comment by YaqiWang** [comment]
It has been merged into yak repo. But it seems this defect has been fixed and the only issue now is that #8162, i.e. the subapp does not inherit the master file base specified in command line.

**Comment by YaqiWang** [comment]
@permcody you can find the test under yak master. Currently I have to specify `Outputs/file_base` for all subapps, could you verify for me with your fix those CLI args are not needed?

**Comment by permcody** [comment]
Looks like #8296 fixes this. Here's the output after I ran that test after removing all of the individual multiapp Outputs overrides:
```
-rw-r--r--   1 permcj  INEL-NT\Domain Users  40620 Jan 10 14:22 nofsrc_out.e
-rw-r--r--   1 permcj  INEL-NT\Domain Users  39848 Jan 10 14:22 master_simplified_out_slave2.i0.e
-rw-r--r--   1 permcj  INEL-NT\Domain Users  39848 Jan 10 14:22 master_simplified_out_slave1.i0.e
-rw-r--r--   1 permcj  INEL-NT\Domain Users  39848 Jan 10 14:22 master_simplified_out_aslave2.i0.e
-rw-r--r--   1 permcj  INEL-NT\Domain Users  39848 Jan 10 14:22 master_simplified_out_aslave1.i0.e
-rw-r--r--   1 permcj  INEL-NT\Domain Users  36008 Jan 10 14:22 master_simplified_out_adjoint.i0.e
```

**Comment by permcody** [comment]
Fixed after merging.

**Comment by YaqiWang** [comment]
Should the subapp output also use the new file base like `nofsrc_out_slave2.i0.e`? Otherwise their output file names will be the same as those without master file base.

**Comment by permcody** [comment]
Well I think that's part of the reason we are in this mess. We didn't give a lot of thought to the CLI system when Multiapps was added. I only came back and dealt with it later. I was trying to maintain lots of backwards compatibility so I kind of made "most" overrides global. I think that's a bad idea in the general case. With this patch, now everything will be separate, no exceptions. You can still override all subapps at once with the `:` syntax but that's it.

Your suggestion here is good, but it's definitely a special case. What you want is a nice pattern so that you can change the whole hierarchy of names at once but have them relative to one and other. Yes this is a good idea but will require some special processing. I think this fix can go in as-is and we can work on implementing a better file_base "tree" override in a separate ticket.

**Comment by YaqiWang** [comment]
Will you create the separate ticket? At least the subapp output will not be swallowed by the CLI file base.

**Comment by permcody** [comment]
Closed by #8286
